driver/t5layernorm_driver.hpp

```diff
@@ -0,0 +1,650 @@
+#include "InputFlags.hpp"
+#include "driver.hpp"
+#include "random.hpp"
+#include "tensor_driver.hpp"
+#include "timer.hpp"
+#include <algorithm>
+#include <cfloat>
+#include <cstdlib>
+#include <memory>
+#include <miopen/tensor.hpp>
+#include <numeric>
+#include <vector>
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloT5LayerNormForwardRunHost(miopenTensorDescriptor_t xDesc,
+                                     Tgpu* x,
+                                     Tgpu* weight,
+                                     Tcheck* yhost,
+                                     Tcheck* rstdhost,
+                                     float eps,
+                                     miopenNormMode_t mode)
+{
+    auto dims         = miopen::deref(xDesc).GetLengths();
+    size_t outer_size = 1;
+    size_t inner_size = dims[dims.size() - 1];
+
+    for(size_t i = 0ULL; i < dims.size() - 1; ++i)
+    {
+        outer_size *= dims[i];
+    }
+
+    int32_t ret = 0;
+
+    for(int32_t o = 0; o < outer_size; o++)
+    {
+        Tcheck pvar = static_cast<Tcheck>(0);
+        for(int32_t i = 0; i < inner_size; i++)
+        {
+            Tcheck tmp = static_cast<Tcheck>(x[o * inner_size + i]);
+            pvar += tmp * tmp;
+        }
+
+        pvar         = pvar / inner_size;
+        Tcheck prstd = static_cast<Tcheck>(1.0) / sqrt(pvar + eps);
+
+        rstdhost[o] = prstd;
+
+        for(int32_t i = 0; i < inner_size; i++)
+        {
+            Tcheck pweight = mode ? static_cast<Tcheck>(weight[i]) : static_cast<Tcheck>(1);
+            yhost[o * inner_size + i] =
+                (static_cast<Tcheck>(x[o * inner_size + i])) * prstd * pweight;
+        }
+    }
+    return ret;
+}
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloT5LayerNormBackwardRunHost(miopenTensorDescriptor_t dyDesc,
+                                      Tgpu* dy,
+                                      Tgpu* x,
+                                      Tgpu* weight,
+                                      Tcheck* rstdhost,
+                                      Tcheck* dxhost,
+                                      miopenNormMode_t mode)
+{
+    auto dims         = miopen::deref(dyDesc).GetLengths();
+    size_t outer_size = 1;
+    size_t inner_size = dims[dims.size() - 1];
+
+    for(size_t i = 0ULL; i < dims.size() - 1; ++i)
+    {
+        outer_size *= dims[i];
+    }
+
+    int32_t ret = 0;
+
+    for(int32_t o = 0; o < outer_size; o++)
+    {
+        Tcheck sum = static_cast<Tcheck>(0);
+        for(int32_t i = 0; i < inner_size; i++)
+        {
+            Tcheck pweight = mode ? static_cast<Tcheck>(weight[i]) : static_cast<Tcheck>(1);
+            Tcheck pdy = dy ? static_cast<Tcheck>(dy[o * inner_size + i]) : static_cast<Tcheck>(0);
+            Tcheck px  = static_cast<Tcheck>(x[o * inner_size + i]);
+            sum += pdy * px * pweight;
+        }
+
+        Tcheck ds    = sum;
+        Tcheck s     = static_cast<Tcheck>(1) / inner_size;
```

Comment:
inner_size has a potential div0 here (see https://github.com/ROCm/MIOpen/pull/2833#discussion_r1573118627)

Reply:
Added to check where input is received.